src/components/textBlocks/ListItem.vue

```diff
@@ -0,0 +1,395 @@
+
+			// Count occurrences of display names in order to provide a distinguishable description if needed
+			const nameCounts = allSuggestions.reduce((nameCounts, result) => {
+				if (!result.displayName) {
+					return nameCounts
+				}
+				if (!nameCounts[result.displayName]) {
+					nameCounts[result.displayName] = 0
+				}
+				nameCounts[result.displayName]++
+				return nameCounts
+			}, {})
+
+			this.suggestions = allSuggestions.map(item => {
+				// Make sure that items with duplicate displayName get the shareWith applied as a description
+				if (nameCounts[item.displayName] > 1 && !item.desc) {
+					return { ...item, desc: item.shareWithDisplayNameUnique }
+				}
+				return item
+			})
+
+			this.loading = false
+			console.info('suggestions', this.suggestions)
+		},
+		/**
+		 * Get the icon based on the share type
+		 *
+		 * @param {number} type the share type
+		 * @return {string} the icon class
+		 */
+		 shareTypeToIcon(type) {
+			switch (type) {
+			case ShareType.User:
+				// default is a user, other icons are here to differentiate
+				// themselves from it, so let's not display the user icon
+				// case this.SHARE_TYPES.SHARE_TYPE_REMOTE:
+				// case this.SHARE_TYPES.SHARE_TYPE_USER:
+				return {
+					icon: 'icon-user',
+					iconTitle: t('files_sharing', 'Guest'),
+				}
+			case ShareType.Group:
+				return {
+					icon: 'icon-group',
+					iconTitle: t('files_sharing', 'Group'),
+				}
+			default:
+				return {}
+			}
+		},
+
+		/**
+		 * Format shares for the multiselect options
+		 *
+		 * @param {object} result select entry item
+		 * @return {object}
+		 */
+		formatForMultiselect(result) {
+			return {
+				shareWith: result.value.shareWith,
+				shareType: result.value.shareType,
+				user: result.uuid || result.value.shareWith,
+				isNoUser: result.value.shareType !== ShareType.User,
+				displayName: result.name || result.label,
+				shareWithDisplayNameUnique: result.shareWithDisplayNameUnique || '',
+				...this.shareTypeToIcon(result.value.shareType),
+			}
+		},
+		/**
+		 * Debounce getSuggestions
+		 *
+		 * @param {...*} args the arguments
+		 */
+		 debounceGetSuggestions: debounce(300, function(...args) {
+			this.getSuggestions(...args)
+		}),
+		handleListItemClick() {
+			if (!this.isViewMode) {
+				this.editModalOpen = true
+				return
+			}
+			this.$emit('click', this.textBlock)
+		},
+		closeTextBlockDialog() {
+			this.editModalOpen = false
+			this.localTextBlock = Object.assign({}, this.textBlock)
+		},
+		newTextBlock() {
+			this.saveLoading = true
+			this.mainStore.patchTextBlock(this.localTextBlock).then(() => {
+				this.saveLoading = false
+				this.editModalOpen = false
+				this.localTextBlock = Object.assign({}, this.textBlock)
+			}).catch(() => {
+				showError(t('mail', 'Failed to save textBlock'))
+			})
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+.textBlock-list-item{
```

Comment:
camel case seems unusual for class names, no? also no space before the opening brackets